realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -266,6 +306,13 @@ public boolean isReadOnly() {
         return readOnly;
     }
 
+    /**
+     * @return {@code true} if this configuration is intended to open a synced Realm offline (example for client reset), {@code false} otherwise.
+     */
+    public boolean isForceSyncHistory() {
```

Comment:
need to be public?

Reply:
yes since it will be accessed from `OsRealmConfig` (internal)

Reply:
Could we call it something else that would make a little more sense to end users in that case? like isForcedOfflineCould we call it something else that would make a little more sense to end users in that case? like isForcedOffline

Reply:
I'm honestly understanding client reset less and less :D but I agree I don't really understand what this does. 

Is it like, `shouldOpenOffline`?

Reply:
I was trying to avoid anything _offline_ on a RealmConfiguration, since the concept applies to Sync only ...

Reply:
Good point, but then `Sync` also only applies to sync only :)

I don't know, it will not be ideal no matter what we do, so I guess this is fine.

Reply:
What about 
```Java
    /**
     * @return {@code true} if this configuration is intended to open the client reset backup Realm
     * {@code false} otherwise.
     *
     * @see <a href="https://realm.io/docs/java/latest/api/io/realm/ClientResetRequiredError.html">ClientResetRequiredError</a>
     */
    public boolean isClientResetBackupRealm() {
    }

```
this is the only use case where we explicitly set the `force_sync_history` to `true` in order to open the Realm without a `SyncConfig`. It will make more sense IMO to the end user, WDYT?